src/context/authStore.tsx

```diff
@@ -88,89 +134,85 @@ export const AuthStoreContextProvider: FunctionComponent<{
     setAuthCredentialsMap({});
   }, []);
 
-  const [, setState] = useState();
   useEffect(() => {
-    const migrateOldAuthFromStore = async (): Promise<boolean> => {
-      const values = await AsyncStorage.multiGet([
-        SESSION_TOKEN_KEY,
-        EXPIRY_KEY,
-        ENDPOINT_KEY,
-      ]);
-      const [sessionToken, expiry, endpoint] = values.map((value) => value[1]);
-      if (sessionToken && endpoint && expiry) {
-        await AsyncStorage.multiRemove([
-          SESSION_TOKEN_KEY,
-          EXPIRY_KEY,
-          ENDPOINT_KEY,
-        ]);
-        const newAuthCredentials = {
-          [endpoint]: {
-            endpoint,
-            sessionToken,
-            expiry: Number(expiry),
-            operatorToken: "",
-          },
-        };
-        setAuthCredentialsMap(newAuthCredentials);
-        AsyncStorage.setItem(
-          AUTH_CREDENTIALS_STORE_KEY,
-          JSON.stringify(newAuthCredentials)
+    /**
+     * Migrates credentials from old auth store to new auth store. Checks stores in order of
+     * increasing age (although now only has v2 to check). Takes the value in the highest version. Clears all
+     * old storage locations.
+     * @param newStorageHasData whether most recent data has been found. If this is true, does not
+     * attempt any migration and just clears old storage locations
+     * @returns the updated value of {@link hasUpdatedData}, i.e. true if any updated data was found
+     * in the older storage (if {@link newStorageHasData} was passed in as true, it will always return true)
+     */
+    const migrateOldAuthFromStore = async (
+      newStorageHasData: boolean
+    ): Promise<boolean> => {
+      let hasUpdatedData = newStorageHasData;
+      // check v2 storage
+      if (!hasUpdatedData) {
+        const authCredentialsString = await AsyncStorage.getItem(
+          AUTH_CREDENTIALS_STORE_KEY
         );
-        return true;
-      } else {
-        return false;
+        if (authCredentialsString) {
+          try {
+            const authCredentialsFromStore: AuthCredentialsMap = JSON.parse(
+              authCredentialsString
+            );
+            setAuthCredentialsMap(authCredentialsFromStore);
+            hasUpdatedData = true; // should this be set b4 parsing? error recovery policy
```

Comment:
Can remove the comment here (:

Reply:
> This is a good point, if the user's existing auth credentials is somehow un-parseable, then they'll be stuck in a loop of migration. I think this is OK because it could have been an error we made. In such a scenario, we can push an update and no data will be lost. If we move this up to before the parsing, then we may accidentally wipe the async storage.

I was thinking about this and realised, even if the data is not lost, once the user adds a new credential, it will be saved to SecureStore. On the next start up of the app, AuthStore will detect that SecureStore has data and wipe the AsyncStorage too. Is that acceptable?